### PR TITLE
When content-length is 0, don't try to collect a response body.

### PR DESCRIPTION
**Pull request by erinishimoticha**

### CODE REVIEW

File: request.js [hunk 1]
```diff
@@ -92,7 +92,19 @@ function request(requestOptions, callback) {
   debug(requestId, "calledBack:", calledBack)
   var req = transport.request(options, function onResponse(res) {
     metadata = res
-    res.pipe(concat(collect))
+
+    var contentLength
+    try {
+      contentLength = parseInt(res.headers['content-length'], 10)
```
Comment (brycebaril) [inline request.js]: I don't think parseInt can throw, but it can return NaN, the `=== 0` check below should probably also check for `NaN`?

**Review by brycebaril — CHANGES_REQUESTED** [review]

File: request.js (hunk 1 above)
Comment (erinishimoticha) [reply request.js]: Oops, you're right.

Comment (erinishimoticha) [reply request.js]: Considering `NaN` to be the same as `Content-Length: 0` messes up other tests. Is it a likely scenario that a user agent would send no `Content-Length` header at all? At worst, if we only check for `contentLength === 0`, the current behavior that this PR addresses will be retained.

Comment (brycebaril) [reply request.js]: Good point, I think it's likely enough that no `Content-Length` header would be sent, but in that case it should probably be assumed that there will probably be content, it's just of indeterminate length. (Such a scenario is common with streaming endpoints, not that this library is designed to support streaming endpoints.)

Comment (erinishimoticha) [reply request.js]: Pushed changes.

**Review by brycebaril — APPROVED** [review]

**Comment by brycebaril** [comment]
@erinspice thanks! Published as `client-request@1.2.1`

**Pull request merged**